builder/vsphere/driver/vm_cdrom.go

```diff
@@ -55,19 +55,24 @@ func (vm *VirtualMachineDriver) CreateCdrom(c *types.VirtualController) (*types.
 	return device, nil
 }
 
-func (vm *VirtualMachineDriver) RemoveCdroms() error {
+func (vm *VirtualMachineDriver) RemoveCdroms(nCdroms int) error {
 	devices, err := vm.Devices()
 	if err != nil {
 		return err
 	}
 	cdroms := devices.SelectByType((*types.VirtualCdrom)(nil))
+	if nCdroms != 0 {
+		cdroms = cdroms[:nCdroms]
+	}
 	if err = vm.RemoveDevice(true, cdroms...); err != nil {
 		return err
 	}
 
-	sata := devices.SelectByType((*types.VirtualAHCIController)(nil))
-	if err = vm.RemoveDevice(true, sata...); err != nil {
-		return err
+	if len(cdroms) > nCdroms {
```

Comment:
I don't understand why this check. How is this removal different from the removal happening above. I see there are different devices types but I don't understand why it only happens if cdroms is greater than nCdroms which will only occur if nCdroms is 0. Is that right?

Reply:
Hmmm… I think that might be a mistake… I'll take a closer look tomorrow

Reply:
Ah, I got it. What this `if` does is it basically checks if we remove all cdroms or not *(that is due to `nCdroms` having special value `0`)*.

Yeah, that's confusing. Anyway, I'll follow with your suggestion below of having a separate function for this, that should resolve the readability of this code as well.

Reply:
The code was removed

Reply:
Thanks for the explanation.